flink-core/src/main/java/org/apache/flink/api/common/ExecutionConfig.java

```diff
@@ -150,6 +150,9 @@
 	/** This flag defines if we use compression for the state snapshot data or not. Default: false */
 	private boolean useSnapshotCompression = false;
 
+	/** Determines if a task fails or not if there is an error in writing its checkpoint data. Default: true */
```

Comment:
Duplicate option in `ExecutionConfig` and `CheckpointingConfig`.

Reply:
After discussions with @aljoscha, we decided to keep this because it is the current way that Flink forwards the configuration. We add some `@Internal` and `@Deprecated` annotations to the methods in `ExecutionConfig` so that the user will use the proper calls on `CheckpointingConfig` instead.